model/RdsResultStorage.php

```diff
@@ -474,27 +481,67 @@ public function getAllDeliveryIds()
     public function getResultByDelivery($delivery, $options = array())
     {
         $returnValue = array();
-        $sql = 'SELECT * FROM ' . self::RESULTS_TABLENAME;
         $params = array();
 
 
-        if (count($delivery) > 0) {
+        $ids = [];
+        if (array_key_exists('startfrom', $options) || array_key_exists('endfrom', $options)
```

Comment:
`if (count(array_intersect(['startfrom', 'endfrom', 'startto', 'endto'], $options))>0){}`
also, pls, use constants for the static data ('startfrom' etc)